src/components/logs/logsWebview.ts

```diff
@@ -61,11 +61,65 @@ export class LogsPanel extends WebPanel {
                 </div>
             </div>
             <script>
-              var lastMode = '';
-              var lastRegexp = '';
-              var renderNonce = 0;
+              let renderNonce = 0;
+              let orig = \`${this.content}\`.split('\\n');
 
-              var orig = \`${this.content}\`.split('\\n');
+              const getFilteredContent = (textToAppend) => {   
+                const regexp = document.getElementById('regexp').value;
+                const mode = document.getElementById('mode').value;                         
+                const rawContent = textToAppend ? textToAppend : orig;
```

Comment:
So if `textToAppend` is passed then it _only_ applies to the new text, but otherwise it applies a new filter to the existing set of logs?  I find this combination of behaviours a bit hard to keep straight in my head... what if we had two functions `filterAll()` (operates on global `content`, for use when applying a new filter) and `filterNewLogs(logsText)` (operates on passed text, for use when appending new logs with an existing filter).  Both could call down to a common function e.g. `filter` but would differ in how they prepared the input to `filter`.